flink-state-backends/flink-statebackend-heap-spillable/src/main/java/org/apache/flink/runtime/state/heap/CopyOnWriteSkipListStateMap.java

```diff
@@ -1367,6 +1317,14 @@ public Long next() {
 		}
 	}
 
+	private Tuple2<ByteBuffer, Integer> getNodeByteBufferAndOffset(long node) {
```

Comment:
ok, the change was following [previous review comment](https://github.com/apache/flink/pull/9501#discussion_r320370445) and let me further improve it. However, I'd like to point out again that such operations are on the core path and will be invoked frequently, which makes the creation of temporary objects a real burden for GC. Now I cannot tell the detailed performance impact or number since it's still half way (or less) upstreamed, but I will keep an eye on the cost of such refactor.

Reply:
Fair enough. I think it is important to keep an eye on the performance. Whether a `Tuple2` or a bit more expressive class is instantiated should not matter much, though.